src/analyseReport.js

```diff
@@ -0,0 +1,106 @@
+const fs = require("fs");
+
+let specs = [];
+const browsers = ["chrome", "firefox"];
+const defaultBrowser = "chrome"; // I will use it in case of there is no browser in the title.
+
+function getBrowserFromTitle(title) {
+  title = title.toLowerCase();
+  for (let browser of browsers) {
+    if (title.includes(browser)) {
+      return browser;
+    }
+  }
+}
+
+function orderBasedOnBrowserDuration(browser) {
+  let aDuration, bDuration;
+  return specs.sort(function (a, b) {
+    for (let i = 0; i < a.data.length; i++) {
+      if (a.data[i].browser === browser) {
+        aDuration = a.data[i].duration;
+        break;
+      }
+    }
+
+    for (let i = 0; i < b.data.length; i++) {
+      if (b.data[i].browser === browser) {
+        bDuration = b.data[i].duration;
+        break;
+      }
+    }
+
+    return aDuration - bDuration;
+  });
+}
+
+function intiateSpecData(specName) {
+  for (let spec of specs) {
+    if (spec["specName"] === specName) {
+      return;
+    }
+  }
+  specs.push({
+    specName: specName,
+    data: browsers.map((browser) => {
+      return { browser: browser, duration: 0 };
+    }),
+  });
+}
+
+function updateSpecData(suites, specName) {
+  for (let suite of suites) {
```

Comment:
`for (let suite of suites)` could be replaced by `suites.forEach(suite => { } )`

Reply:
done